### DISCUSSION TITLE
Fragment ions assigned a precursor(s)

### DISCUSSION THREAD

**Discussion by ABB-808** [vdemichev/DiaNN#1141] [Q&A] [answered]
Hi- 

I have noticed when opening the report.tsv or the report.parquet with --export-quant enabled that there seems to always be at most 12 fragment ions ID'd for each precursor. report.parquet has column names Fr.0.ID - Fr.11.ID as well as score, index and quantity. report.tsv had fragment ions intensities of at most 12 fragment ions. Is the cutoff for number of precursor fragment ions 12? Is this always the case or can this be changed and is there any reason to do so?

**Comment by vdemichev** [comment]
Hi,

This is the number of fragments used for quantification. The reason for this is that using extra would not really improve quantities: if out of top 12 library fragments few or none produce good signals, the furhter (less abundant) fragments are unlikely to be better.

Best,
Vadim

**Reply by ABB-808** [reply]
Thanks for getting back to me. 

So the 12 fragment ions shown are just the ones used for quantification. Is the number of fragment ions used to identify a precursor ever recorded in an output file? The reason I ask is that I am curious about the quality of my spectral data and want to have a readout for how "useful" each ms2 scan is (are there enough ions present? are there too many ions present? etc.)

**Reply by vdemichev** [reply, marked as answer]
All library fragments are used for identification.
For such a readout, DIA-NN Viewer (and the associated XIC-saving) could be great, i.e. shows exactly what DIA-NN sees.
